examples/features/authz/server/main.go

```diff
@@ -0,0 +1,255 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Binary server is an example server.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/authz"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/examples/data"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	pb "google.golang.org/grpc/examples/features/proto/echo"
+)
+
+const (
+	unaryEchoWriterRole      = "UNARY_ECHO:W"
+	streamEchoReaderRole     = "STREAM_ECHO:R"
+	streamEchoWriterRole     = "STREAM_ECHO:W"
+	streamEchoReadWriterRole = "STREAM_ECHO:RW"
+)
+
+var (
+	port = flag.Int("port", 50051, "the port to serve on")
+
+	errMissingMetadata = status.Errorf(codes.InvalidArgument, "missing metadata")
+	errInvalidToken    = status.Errorf(codes.Unauthenticated, "invalid token")
+
+	authzPolicy = newPolicy("authz",
+		[]string{
+			newHeaderRule("allow_UnaryEcho", "/grpc.examples.echo.Echo/UnaryEcho", unaryEchoWriterRole),
+			newHeaderRule("allow_ClientStreamingEcho", "/grpc.examples.echo.Echo/ClientStreamingEcho", streamEchoWriterRole),
+			newHeaderRule("allow_ServerStreamingEcho", "/grpc.examples.echo.Echo/ServerStreamingEcho", streamEchoReaderRole),
+			newHeaderRule("allow_BidirectionalStreamingEcho", "/grpc.examples.echo.Echo/BidirectionalStreamingEcho", streamEchoReadWriterRole),
+		},
+		nil,
+	)
+	mockedMetadata = newMockedMetadata()
+)
+
+func newMockedMetadata() metadata.MD {
+	md := metadata.MD{}
+	roles := []string{
+		unaryEchoWriterRole,
+		streamEchoReadWriterRole,
+	}
+	for _, role := range roles {
+		md.Set(role, "true")
+	}
+	return md
+}
+
+func newPolicy(name string, allowRules, denyRules []string) string {
+	return fmt.Sprintf(`{
+		"name": "%s",
+		"allow_rules":[%s],
+		"deny_rules":[%s]
+	}`, name,
+		strings.Join(allowRules, ","),
+		strings.Join(denyRules, ","),
+	)
+}
+
+func newHeaderRule(name, path, role string) string {
+	return fmt.Sprintf(`{
+		"name": "%s",
+		"request": {
+			"paths":["%s"],
+			"headers": [
+				{
+					"key": "%s",
+					"values": ["true"]
+				}
+			]
+		}
+	}`, name, path, role)
+}
```

Comment:
I would prefer specifying the policy as a string instead of constructing it programmatically, for the following reasons:
- this is an example, and if we can reduce the amount of code someone has to read to understand this feature, its a win
- the structure of the policy will be more evident

Reply:
Replaced for constant string